dotnet/src/Connectors/Connectors.Memory.Redis/RedisVectorStore.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Microsoft.SemanticKernel.Data;
+using NRedisStack.RedisStackCommands;
+using StackExchange.Redis;
+
+namespace Microsoft.SemanticKernel.Connectors.Redis;
+
+/// <summary>
+/// Class for accessing the list of collections in a Redis vector store.
+/// </summary>
+/// <remarks>
+/// This class can be used with collections of any schema type, but requires you to provide schema information when getting a collection.
+/// </remarks>
+public sealed class RedisVectorStore : IVectorStore
+{
+    /// <summary>The name of this database for telemetry purposes.</summary>
+    private const string DatabaseName = "Redis";
+
+    /// <summary>The redis database to read/write indices from.</summary>
+    private readonly IDatabase _database;
+
+    /// <summary>Optional configuration options for this class.</summary>
+    private readonly RedisVectorStoreOptions _options;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="RedisVectorStore"/> class.
+    /// </summary>
+    /// <param name="database">The redis database to read/write indices from.</param>
+    /// <param name="options">Optional configuration options for this class.</param>
+    public RedisVectorStore(IDatabase database, RedisVectorStoreOptions? options = default)
+    {
+        Verify.NotNull(database);
+
+        this._database = database;
+        this._options = options ?? new RedisVectorStoreOptions();
+    }
+
+    /// <inheritdoc />
+    public IVectorStoreRecordCollection<TKey, TRecord> GetCollection<TKey, TRecord>(string name, VectorStoreRecordDefinition? vectorStoreRecordDefinition = null) where TRecord : class
+    {
+        if (typeof(TKey) != typeof(string))
+        {
+            throw new NotSupportedException("Only string keys are supported.");
+        }
```

Comment:
Did we consider making `IVectorStore` as `IVectorStore<TKey>`, so in this case it will be `RedisVectorStore : IVectorStore<string>`? 

Because having generic method which works only with one type may be confusing.

Reply:
Mmm, we didn't consider it no.  Let's chat offline, since I think there are pros and cons.